src/distributions/range.rs

```diff
@@ -119,8 +126,8 @@ pub trait SampleRange: PartialOrd+Sized {
 /// let range = Range::new(low, high);
 /// let x = range.sample(&mut thread_rng());
 /// ```
-pub trait RangeImpl {
-    /// The type sampled by this implementation (output type).
+pub trait RangeImpl: Sized {
+    /// The type sampled by this implementation.
     type X: PartialOrd;
```

Comment:
Why does this type have a `PartialOrd` bound?

Reply:
Um... it might be to allow the `low < high` check. You can try removing it if you like.

Reply:
Now that I looked a bit better at it, `SampleRange` already has `PartialOrd+Sized` as trait bounds, which takes care of the `low < high` check. But the extra bound kind of makes sense: you can't speak about a range between two points if there is not some sort of order between the elements. I will leave it for now.